src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/securityrule/AzureSecurityRulePlugin.java

```diff
@@ -0,0 +1,130 @@
+            throws FogbowException {
+
+        LOGGER.info(Messages.Info.REQUESTING_INSTANCE_FROM_PROVIDER);
+        checkOrderType(majorOrder);
+
+        String networkSecurityGroupName = AzureGeneralUtil.defineInstanceId(majorOrder.getInstanceId());
+        String networkSecurityGroupId = AzureResourceIdBuilder.networkSecurityGroupId()
+                .withSubscriptionId(azureUser.getSubscriptionId())
+                .withResourceGroupName(this.defaultResourceGroupName)
+                .withResourceName(networkSecurityGroupName)
+                .build();
+        String cidr = securityRule.getCidr();
+        int portFrom = securityRule.getPortFrom();
+        int portTo = securityRule.getPortTo();
+        SecurityRule.Direction direction = securityRule.getDirection();
+        SecurityRule.Protocol protocol = securityRule.getProtocol();
+        String securityRuleName = AzureGeneralUtil.generateResourceName();
+
+        AzureUpdateNetworkSecurityGroupRef azureUpdateNetworkSecurityRef = AzureUpdateNetworkSecurityGroupRef.builder()
+                .ruleResourceName(securityRuleName)
+                .networkSecurityGroupId(networkSecurityGroupId)
+                .protocol(protocol)
+                .cidr(cidr)
+                .direction(direction)
+                .portFrom(portFrom)
+                .portTo(portTo)
+                .checkAndBuild();
+
+        this.azureNetworkSecurityGroupOperationSDK.doCreateInstance(azureUpdateNetworkSecurityRef, azureUser);
+
+        return SecurityRuleIdContext.buildInstanceId(networkSecurityGroupName, securityRuleName);
+    }
+
+    @Override
+    public List<SecurityRuleInstance> getSecurityRules(Order majorOrder, AzureUser azureUser) throws FogbowException {
+        LOGGER.info(String.format(Messages.Info.GETTING_INSTANCE_S, majorOrder.getInstanceId()));
+
+        return doGetSecurityRules(majorOrder, azureUser);
+    }
+
+    @Override
+    public void deleteSecurityRule(String securityRuleId, AzureUser azureUser) throws FogbowException {
+        LOGGER.info(String.format(Messages.Info.DELETING_INSTANCE_S, securityRuleId));
+
+        SecurityRuleIdContext securityRuleIdContext = new SecurityRuleIdContext(securityRuleId);
+
+        String networkSecurityGroupName = securityRuleIdContext.getNetworkSecurityGroupName();
+        String networkSecurityGroupId = AzureResourceIdBuilder.networkSecurityGroupId()
+                .withSubscriptionId(azureUser.getSubscriptionId())
+                .withResourceGroupName(this.defaultResourceGroupName)
+                .withResourceName(networkSecurityGroupName)
+                .build();
+
+        String securityRuleName = securityRuleIdContext.getSecurityRuleName();
+        this.azureNetworkSecurityGroupOperationSDK.deleteNetworkSecurityRule(networkSecurityGroupId, securityRuleName, azureUser);
+    }
+
+    @VisibleForTesting
+    List<SecurityRuleInstance> doGetSecurityRules(Order majorOrder, AzureUser azureUser)
```

Comment:
In this case, we have the same approuch to both majorOrders. Thus, this method needs to behave same way to both orders so, it is not necessary the "swithcase".

Reply:
Done.